srt-protocol/src/protocol/sender/mod.rs

```diff
@@ -233,11 +224,18 @@ impl Sender {
         //      1).
 
         //   3) Wait until there is application data to be sent.
-        else if self.transmit_buffer.is_empty() && !self.close_requested {
+        else if self.transmit_buffer.is_empty() {
             // TODO: the spec for 3) seems to suggest waiting at here for data,
             //       but if execution doesn't jump back to Step1, then many of
             //       the tests don't pass... WAT?
-            return WaitForData;
+
+            if self.status.should_drain() && self.send_buffer.len() == 1 {
+                if let Some(packet) = self.send_buffer.pop() {
+                    self.send_data(packet, now);
+                    self.lr_acked_packet = self.transmit_buffer.next_sequence_number;
```

Comment:
Again this hack could use at least a comment. Maybe we can think of a better way to handle this?

Reply:
I struggled with this a bit too, the way I see it, there are two options:
1) avoid modifying state directly to arrive at a flush state, but rather have SND responsible for explicitly transitioning to a "flushed" state by updating the status field, so is_flushed would merely check to see if the sender is in a flushed state
2) extract method on this draining flush logic to document the intent of the ugly better